nil/internal/types/code.go

```diff
@@ -22,7 +25,7 @@ func (c Code) Clone() Code {
 
 func (c Code) Hash() common.Hash {
 	if len(c) == 0 {
-		return common.EmptyHash
+		return EmptyCodeHash
 	}
```

Comment:
Good catch!

Maybe we can drop this condition at all because `common.KeccakHash` from empty array should be exactly EmptyCodeHash